source/vof/reconstruct.cc

```diff
@@ -0,0 +1,389 @@
+/*
+ Copyright (C) 2016 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/global.h>
+#include <aspect/simulator.h>
+#include <aspect/vof/handler.h>
+#include <aspect/vof/utilities.h>
+
+namespace aspect
+{
+  using namespace dealii;
+
+  template <>
+  void VoFHandler<2>::update_vof_normals (const VoFField<2> field,
+                                          LinearAlgebra::BlockVector &solution)
+  {
+    const int dim = 2;
+
+    LinearAlgebra::BlockVector initial_solution;
+
+    sim.computing_timer.enter_section("   Reconstruct VoF interfaces");
+
+    initial_solution.reinit(sim.system_rhs, false);
+
+    // Boundary reference
+    typename DoFHandler<dim>::active_cell_iterator endc =
+      sim.dof_handler.end ();
```

Comment:
I know this class has direct access to the simulator, but I would still ask you to replace `sim.` by `this->get_...` wherever possible. There are three reasons: First, it is simpler for others to read, because most of the code base uses SimulatorAccess to access Simulator variables. Second, it is simpler to find out which calls actually change members of Simulator. Third, it is safer, because you can not accidentally change things.